tools/osbuilder/image-builder/image_builder.sh

```diff
@@ -138,6 +141,7 @@ build_with_container() {
 	image_name=$(basename "${image}")
 
 	"${container_engine}" build  \
+		   --build-arg IMAGE_REGISTRY="${IMAGE_REGISTRY}" \
```

Comment:
Hi @egernst, wouldn't make sense to pass the argument only and only if IMAGE_REGISTRY is exported by the user? Otherwise it will always overwrite the default in Dockerfile.

Reply:
Yeah, agreed, I was just looking at that. In that case, we'd want to drop the default at top of file, and then do something like:

```bash
               REGISTRY_ARG=""
+               if [ -n "${IMAGE_REGISTRY}" ]; then
+                       echo "it was set"
+                       REGISTRY_ARG="--build-arg IMAGE_REGISTRY=${IMAGE_REGISTRY}"
+               fi
+
                # setup to install go or rust here
                generate_dockerfile "${distro_config_dir}"
                "$container_engine" build  \
-                       --build-arg IMAGE_REGISTRY="${IMAGE_REGISTRY}" \
+                       ${REGISTRY_ARG} \
                        --build-arg http_proxy="${http_proxy}" \
                        --build-arg https_proxy="${https_proxy}" \
                        -t "${image_name}" "${distro_config_dir}"
 ```